### PR TITLE
feat: modernize code

### PR DESCRIPTION
**Pull request by rabyyuson**
### Tickets

https://horizenlabs.atlassian.net/browse/TOOLS-2322

### Changes

- modernize codebase (modern node/jest/typescript version support)
- remove old code references and incrementally build the RPC methods tests
- include the following tests: `eth_blockNumber`, `eth_getBalance`, and `eth_gasPrice`

### Preview

https://github.com/HorizenOfficial/rpc-tests/assets/5024180/e633ebe2-b755-4204-8c91-496433674faf

### CODE REVIEW

**Comment by rabyyuson** [comment]
@yurikoinaba @voyenavion what do you both think about this pull request?

File: rpc/eth/getBlockByNumber/index.test.ts
[the remark this answers is not in the document]
Comment (rabyyuson) [reply rpc/eth/getBlockByNumber/index.test.ts:1]: yeah definitely something to improve on! I'll add a note on https://horizenlabs.atlassian.net/browse/TOOLS-2323 to refactor the way we're performing the assertions. thanks for the suggestion @voyenavion!

**Review by voyenavion — APPROVED** [review]
lgtm!

File: utils/urlProvider.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const urlProvider = {
+  rpcUrl: "https://gobi-testnet.horizenlabs.io/ethv1",
```
Comment (yurikoinaba) [inline utils/urlProvider.ts:2]: At some point, make this customizable

Comment (rabyyuson) [reply utils/urlProvider.ts:2]: Good point! I added it to the comment for the test RPC ticket https://horizenlabs.atlassian.net/browse/TOOLS-2323 so I can keep track!

**Pull request merged**